packages/core/etc/simplevm/utils/get_metadata.sh

```diff
@@ -44,21 +86,44 @@ if [ -z "${REAL_METADATA_ENDPOINT}" ]; then
 fi
 
 # Fetch the actual metadata from the extracted endpoint
+# LOCAL_IP="192.168.2.37" # adjust for development purposes
+# log_message ${LOCAL_IP}
+
 LOCAL_IP=$(hostname -I | awk '{print $1}')
 metadata_response=$(curl -s -X GET "${REAL_METADATA_ENDPOINT}/metadata/${LOCAL_IP}" -H "${AUTH_HEADER}")
 
 if [ $? -ne 0 ]; then
   log_message "Error: Failed to fetch metadata"
+  reset_interval
   exit 1
 fi
 
+  key_not_found_flag=$(echo "$metadata_response" | jq -r '.detail')
+
+  if [ "$key_not_found_flag" == "Key not found" ]; then
```

Comment:
Would be better  to check for status codes (403,404 etc.) instead of string output